src/Writer/XlsExcelWriter.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Writer;
+
+/**
+ * .xls writer bridge for library `phpoffice/phpexcel`.
+ *
+ * @see https://github.com/PHPOffice/PHPExcel
+ *
+ * @author Lukáš Brzák <lukas.brzak@email.cz>
+ */
+final class XlsExcelWriter implements TypedWriterInterface
+{
+    /** @var \PHPExcel $excel */
+    protected $excel;
+
+    /** @var \PHPExcel_Worksheet $sheet */
+    protected $sheet;
+
+    /** @var \PHPExcel_Writer_Excel2007 $writer */
+    protected $writer;
+
+    /** @var string $column */
+    protected $column = 'A';
+
+    /** @var int $row */
+    protected $row = 1;
+
+    /** @var string $file */
+    protected $file;
+
+    /** @var bool $showHeaders */
+    protected $showHeaders;
+
+    /**
+     * @param string $file
+     * @param bool   $showHeaders
+     *
+     * @throws \PHPExcel_Exception
+     */
+    public function __construct($file = 'php://output', $showHeaders = true)
+    {
+        $this->file = $file;
+        $this->showHeaders = $showHeaders;
+        $this->configure();
+    }
+
+    public function open()
+    {
+        $this->writer = new \PHPExcel_Writer_Excel2007($this->excel);
+    }
+
+    /**
+     * @param array $data
+     *
+     * @throws \PHPExcel_Writer_Exception
+     */
+    public function write(array $data)
+    {
+        $this->init($data);
+
+        foreach ($data as $header => $value) {
+            $this->sheet->setCellValue(
+                $this->getCurrentCell(),
+                $value
+            );
+            $this->nextColumn();
+        }
+        $this->nextRow();
+    }
+
+    /**
+     * Save the file.
+     *
+     * @throws \PHPExcel_Writer_Exception
+     */
+    public function close()
+    {
+        $this->writer->save($this->file);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getDefaultMimeType()
+    {
+        return 'application/vnd.ms-excel';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getFormat()
+    {
+        return 'xls';
```

Comment:
So the lib generate a xls or a xlsx file at this end ? The MimeType is not the same: 

- .xls      application/vnd.ms-excel
- .xlsx     application/vnd.openxmlformats-officedocument.spreadsheetml.sheet